controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +156,68 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	requiredCrds := []string{"VolumeReplicationClass", "VolumeReplication"}
+
+	for _, crd := range requiredCrds {
+		err := r.waitForResource(crd)
+		if err != nil {
+			r.Log.Error(err, "unexpected error occurred while checking CRD", "CRD", crd)
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *VolumeReplicationGroupReconciler) waitForResource(resourceName string) error {
+	result, err := r.resourceExists(resourceName)
+
+	if !result {
+		if err != nil {
+			return err
+		}
+
+		r.Log.Info(fmt.Sprintf("CRD %s is not installed on cluster, waiting until it appears", resourceName))
+
+		for {
+			time.Sleep(crdCheckInterval)
+
+			result, err := r.resourceExists(resourceName)
```

Comment:
Let be consistent with the happy path pattern:

    if err != nil {
        return err
    }

Reply:
Done